src/elements/content-uploader/UploadStateV2.js

```diff
@@ -0,0 +1,112 @@
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadError} />} />;
+            break;
+        case VIEW_UPLOAD_EMPTY:
+            if (!hasItems) {
+                icon = isOver ? <UploadIllustration width={200} height={200} /> : <UploadIllustration />;
+            }
+            /* eslint-disable no-nested-ternary */
+            content =
+                canDrop && hasItems ? (
+                    <UploadStateContent message={<FormattedMessage {...messages.uploadInProgress} />} />
+                ) : isTouch ? (
+                    <UploadStateContent
+                        fileInputLabel={<FormattedMessage {...messages.uploadNoDragDrop} />}
+                        onChange={onSelect}
+                        useButton
+                    />
+                ) : (
+                    <UploadStateContent
+                        fileInputLabel={<FormattedMessage {...messages.uploadEmptyFileInputV2} />}
+                        folderInputLabel={
+                            isFolderUploadEnabled && <FormattedMessage {...messages.uploadEmptyFolderInput} />
+                        }
+                        message={
+                            !Browser.isIE() &&
+                            (isFolderUploadEnabled ? (
+                                <FormattedMessage {...messages.uploadEmptyWithFolderUploadEnabled} />
+                            ) : (
+                                <FormattedMessage {...messages.uploadEmptyWithFolderUploadDisabled} />
+                            ))
+                        }
+                        onChange={onSelect}
+                        useButton
+                    />
+                );
+            /* eslint-enable no-nested-ternary */
+            break;
+        case VIEW_UPLOAD_IN_PROGRESS:
+            icon = <UploadIllustration />;
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadInProgress} />} />;
+            break;
+        case VIEW_UPLOAD_SUCCESS:
+            icon = <UploadSuccessState />;
+            content = (
+                <UploadStateContent
+                    fileInputLabel={<FormattedMessage {...messages.uploadSuccessFileInput} />}
+                    folderInputLabel={
+                        isFolderUploadEnabled && <FormattedMessage {...messages.uploadSuccessFolderInput} />
+                    }
+                    message={<FormattedMessage {...messages.uploadSuccess} />}
+                    onChange={onSelect}
+                    useButton={isTouch}
+                />
+            );
+            break;
+        default:
+            break;
+        /* eslint-enable jsx-a11y/label-has-for */
+    }
+
+    const className = classNames('bcu-upload-state', {
+        'bcu-is-droppable': isOver && canDrop,
+        'bcu-is-not-droppable': isOver && !canDrop,
+        'bcu-has-items': hasItems,
+    });
+
+    return (
+        <div className={className}>
+            <div>
```

Comment:
Should we be looking at more semantic elements for accessibility (if applicable)?

Reply:
Take a look at the Media component